### ISSUE TITLE
[google_maps_flutter] Get duration of animateCamera

### ISSUE THREAD

**Issue by magax** [flutter/flutter#57404]
Is there any way to know the duration of the animation created by`animateCamera()`? Since the Future completes at the start of the animation and not at the end (I don't understand why), if I want to perform some actions based on the new camera position I have to use a `Timer()` but so far I am using an arbitrary `Duration()` of 500ms to be sure that the animation is done.

**Comment by TahaTesser** [comment]
Hi @magax 
This platform is not meant for assistance on personal code. 
Please questions, please see https://flutter.dev/community for resources and asking questions like this, you may also get some help if you post it on Stack Overflow. 
Closing, as this isn't an issue with Flutter itself, if you disagree please write in the comments and I will reopen it 
Thank you

**Comment by magax** [comment]
Yes I'm sorry but I kind of disagree, this issue is related to the fact that `animateCamera()` completes its Future before the camera animation ends. So no action is possible on the final camera position. Except if we use a workaround like the one I've mentioned. I am not looking for further help since my code is already working but I don't think it's a proper way to do, so you guys should consider to fix this one day.
Cheers!

**Comment by xapatjb4** [comment]
Facing the same problem. No way of knowing when animation is complete so had to use an arbitrary time
